external-crates/move/crates/move-binary-format/src/unit_tests/compatibility_tests.rs

```diff
@@ -1199,3 +1200,57 @@ fn check_new_changed_missing_declarations() {
     assert!(InclusionCheck::Subset.check(&empty, &m1).is_ok());
     assert!(InclusionCheck::Equal.check(&empty, &m1).is_err());
 }
+
+fn arbitrary_map() -> impl Strategy<Value = BTreeMap<i32, i32>> {
+    prop::collection::btree_map(0..1000, 0..1000, 0..20)
+}
+
+proptest! {
+    #[test]
+    fn test_compare_ord_iters(old in arbitrary_map(), new in arbitrary_map()) {
+        let result: Vec<_> = compare_ord_iters(old.iter(), new.iter()).collect();
+
+        for mark in &result {
+                match mark {
+                    Mark::New(key, _) => {
+                        // New keys must only be in `new`
+                        assert!(new.contains_key(key));
+                        assert!(!old.contains_key(key));
+                    },
+                    Mark::Missing(key, _) => {
+                        // Missing keys must only be in `old`
+                        assert!(old.contains_key(key));
+                        assert!(!new.contains_key(key));
+                    },
+                    Mark::Existing(key, old_value, new_value) => {
+                        // Existing keys must be in both
+                        assert!(old.contains_key(key));
+                        assert!(new.contains_key(key));
+                        // Values must match the input maps
+                        assert_eq!(Some(old_value), old.get(key).as_ref());
+                        assert_eq!(Some(new_value), new.get(key).as_ref());
+                    },
+                }
+            }
+
+            // Check that all keys in `old` and `new` are represented in the result
+            let mut old_keys: Vec<_> = old.keys().collect();
+            let mut new_keys: Vec<_> = new.keys().collect();
+            old_keys.sort();
+            new_keys.sort();
+
+            let mut combined = old_keys.clone();
```

Comment:
If you use `BTreeSet`s here for `old_keys` and `new_keys` (and for `combined` etc) instead of vectors it may make the logic here simpler.